### ISSUE TITLE
At what level should our sound effects be mastered?

### ISSUE THREAD

**Issue by jbphet** [phetsims/tambo#5]
We are using a number of different sound effects in various sims, and the number is growing as we experiment with sonification.  The loudness of the sound effects varies quite a bit, which makes them challenging to mix in a sim, so I think it would be good for us to standardize on some value.  I took a quick look around on line and found a short article on the topic, see http://www.creativefieldrecording.com/2017/05/10/2-techniques-for-setting-sound-effects-library-mastering-levels/.

I'd like to get input from @Ashton_Morris and @mikewinters10 to see if they have any insights.

**Comment by jbphet** [comment]
I found a posting that sounded pretty reasonable here: http://exchange.designingsound.org/questions/question/what-kind-of-levels-do-you-shoot-for-with-your-own-sfx-library-samples/

In short, this guy recommends using levels that reflect that actual sound level that a listener would hear, and he recommends (and I think these are for the peak level, not RMS):

+ Specifics, loud (ex., metal hits, train passes, door closes, phone rings, gun shots): -10 to -3
+ Specifics, typical (ex., door creaks, footsteps): -12 to -7
+ Specifics, quiet (ex., fridge hum, mouse clicks and movement) -20 to -15
+ Ambiences, loud (ex., traffic, busy crowd): -10 to -3
+ Ambiences, typical (ex., chatting crowd): -15 to -7
+ Ambiences, quiet (ex., soft wind, room tones): -20 to -15. Some room tones may be as low as -25 and be acceptable.
+ I always leave a bit of headroom, and never capture audio right up to 0. That leaves no room for play.

I'm going to target these levels for now.

**Comment by mikewinters10** [comment]
Yes, I also think it would be a good idea to standardize somehow. Those
values seem good to me, assuming they are for dBFS. If we use dB, we should
probably create a 'dbAmp' method that would allow us to easily get a
amplitude value (e.g. 0.dbAmp = 1, -10.dbAmp = 0.316).

Related to this, it seems like it would be a cleaner to normalize all audio
samples on the backend and set the levels in code. That way when we specify
something like ('-20dB), we know that we are referring a fully normalized
sample.

**Comment by jbphet** [comment]
Yes, the values are in dBFS, thanks for bringing that up.  I'll edit my previous comment above to make it more clear.

@mikewinters10 - I'm not quite sure what you mean when you say, "[I]t seems like it would be a cleaner to normalize all audio samples on the backend and set the levels in code."  Can you please clarify?  Are you saying that we should normalize them all to 0 dBFS (or just below)?

**Comment by mikewinters10** [comment]
Sure, what I mean is that instead of having a lot of sound samples with
undetermined amplitude, we could standardize this by normalizing all
sound-samples to 0dBFS (or just below e.g. -3dBFS). That way, when we
specify that a sound clip will have an amplitude of -10dB in one sim, we
will know that if we use another sound clip in a different sim at -10dBFS,
they will have the same amplitude.

**Comment by jbphet** [comment]
So it seems the question is whether to master things at close to the max level so that we have full control in software, or master them at a level that would be close to how loud we would want them in a sim at default volume levels.  I think I'd lean towards the latter, given the blog post that I referenced above, but I'm not wedded to it.  I'll ask @Ashton_Morris too and see if he has any opinion.

**Comment by Ashton-Morris** [comment]
Personally I usually master things at close to the volume that they will eventually be presented so there is less work down the road. But considering we will possibly be using the same sounds in different sims, overhauling old ones in the future, and that we have multiple people that might at some point be touching this then it might be best to stick to a consistent level. 

It will take a bit more work when implementing the sound but at least at any point in the future whoever is working on it or cross using them, then they wont have to question the levels. 

In that case I think we should normalize all audio to -1db. There is something called inter-sample peaking, and mastering at -1db usually leaves room for that to not occur.

**Comment by jbphet** [comment]
I discussed with with @Ashton-Morris a bit more today, and we decided that we will master sounds that are expected to be pretty prominent in the sound design at a level of -1 dbFS, and sounds that are intended to be more subtle or in the background of the sound design will be mastered at a level that is close to the expected playback level.

**Comment by jbphet** [comment]
The plan noted in the previous comment seem reasonable for now, closing, will reopen or create a new issue if we need to revisit this for some reason.

**Comment by jbphet** [comment]
@pixelzoom recently mentioned that there is a document and semi-standard that Apple created called "Mastered for iTunes".  The document is available [here](https://images.apple.com/itunes/mastered-for-itunes/docs/mastered_for_itunes.pdf).  Below is an excerpt that is relevant, I think, to this GitHub issue:

> With digital files, there’s a limit to how loud you can make a track: 0dBFS. Trying to increase a track’s overall loudness beyond this point results in distortion caused by clipping and a loss in dynamic range. The quietest parts of a song increase in volume, yet the louder parts don’t gain loudness due to the upper limits of the digital format.  Although iTunes doesn’t reject files for a specific number of clips, tracks which have audible clipping will not be badged or marketed as Mastered for iTunes. A less obvious issue when setting gain for digital masters can occur on playback. Whether it's a compressed file like an AAC file or an uncompressed file such as a CD, digital data goes through several processes to be converted to an analog signal for playback.

> One common process is called oversampling. This upsamples the digital data at four times the original sample rate to improve the quality of the digital audio signal being converted to analog. If the original digital audio data is at 0dBFS, oversampling can result in undesirable clipping. And if the original was 
 already clipped, oversampling can make it worse. A growing consensus is emerging that digital masters 
should have a small amount of headroom (roughly 1dB) in order to avoid such clipping.

This would seem to lend support to the idea of mastering our sound effects around -1 dBFS and adjusting the gain down as needed.

**Comment by mikewinters10** [comment]
-1dBFS seems like a good choice to me. This means that imported clips will
have the highest possible signal to noise ratio, and will be the loudest
they can be without clipping. If this volume is too loud for a given Sim,
it will be easy enough to master the levels in code.

Mike
